lms/static/scripts/frontend_apps/components/GroupConfigSelector.js

```diff
@@ -106,93 +193,79 @@ export default function GroupConfigSelector({
       useGroupSet: false,
       groupSet: null,
     });
-  };
+  }, [onChangeGroupConfig]);
+
+  const onGroupSelectChange = useCallback(
+    /**
+     * @param {string|null} groupSetID
+     */
+    groupSetID => {
+      onChangeGroupConfig({
+        useGroupSet,
+        groupSet: groupSetID,
+      });
+    },
+    [useGroupSet, onChangeGroupConfig]
+  );
 
   useEffect(() => {
     if (useGroupSet && !haveFetchedGroups) {
       fetchGroupSets();
     }
   }, [fetchGroupSets, useGroupSet, haveFetchedGroups]);
 
+  if (fetchError) {
+    if (fetchError instanceof GroupListEmptyError) {
+      return <NoGroupsError onCancel={onErrorCancel} />;
+    } else if (isAuthorizationError(fetchError)) {
+      return (
+        <AuthorizationModal
+          authURL={/** @type {string} */ (listGroupSetsAPI.authUrl)}
+          authToken={authToken}
+          onAuthComplete={fetchGroupSets}
+          onCancel={onErrorCancel}
+        >
+          <p>Hypothesis needs your permission to show group sets.</p>
+        </AuthorizationModal>
+      );
+    } else {
+      return (
+        <ErrorModal
+          cancelLabel="Cancel"
+          description="There was a problem fetching group sets"
+          error={fetchError}
+          onCancel={onErrorCancel}
+          onRetry={fetchGroupSets}
+        />
+      );
+    }
+  }
+
```

Comment:
The rendering `return` has been restructured such that error states are handled separately. The `NoGroupsError` handling is added here; the other error handling was present before, just moved.